### PR TITLE
Selected page controller is not displayed in the Inspect Panel #2001

### PR DESCRIPTION
**Pull request by ashklianko**
-selector was not setting right value after reload

### CODE REVIEW

File: src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts [hunk 1]
```diff
@@ -176,17 +178,29 @@ export class PageTemplateAndControllerSelector
...
+            if (!isAlreadySelected) {
+                this.setValue(valueToSelectOnReload, true);
+            }
+        } else {
+            this.selectInitialOption();
+        }
+        this.preSelectedValue = null;
+    }
+
+    private getValueToSelectOnReload(): string {
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts:200]: Call it `getPreselectedValue`

File: src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts [hunk 2]
```diff
@@ -176,17 +178,29 @@ export class PageTemplateAndControllerSelector
...
+        return !!this.preSelectedValue ? this.preSelectedValue : this.getValue();
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts:201]: `return this.preselectedValue || this.getValue();`

File: src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts [hunk 3]
```diff
@@ -37,6 +37,8 @@ export class PageTemplateAndControllerSelector
 
     private autoOption: Option<PageTemplateOption>;
 
+    private preSelectedValue: string;
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts:40]: preselectedValue

File: src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts [hunk 4]
```diff
@@ -304,6 +318,8 @@ export class PageTemplateAndControllerSelector
         let optionToSelect = this.getOptionByValue(key);
         if (optionToSelect) {
             this.selectOption(optionToSelect, true);
+        } else {
+            this.preSelectedValue = key;
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts:322]: `this.preSelectedValue`

File: src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts [hunk 5]
```diff
@@ -176,17 +178,29 @@ export class PageTemplateAndControllerSelector
...
-                this.setValue(selectedValue, true);
-            } else {
-                this.selectInitialOption();
-            }
+            this.handleReloaded(templateOptions, controllerOptions);
         }).catch(DefaultErrorHandler.handle);
     }
 
+    private handleReloaded(templateOptions: Option<PageTemplateOption>[], controllerOptions: Option<PageControllerOption>[]) {
+        const valueToSelectOnReload: string = this.getValueToSelectOnReload();
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts:186]: `const preSelectedValue: ...`

File: src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts [hunk 6]
```diff
@@ -176,17 +178,29 @@ export class PageTemplateAndControllerSelector
...
+        this.removeAllOptions();
+        this.initOptionsList(templateOptions, controllerOptions);
+        if (valueToSelectOnReload) {
+            const isAlreadySelected: boolean = !!this.getSelectedOption() && this.getSelectedOption().value === valueToSelectOnReload;
+            if (!isAlreadySelected) {
+                this.setValue(valueToSelectOnReload, true);
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts:192]: Shouldn't we check that `preselectedValue` is still in the list of options after reload?

Comment (ashklianko) [reply src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts:192]: No need, if option is missing nothing will happen

**Pull request merged**